### DISCUSSION TITLE
How to use GraphicsCapturePicker in WinUI 3?  Error when invoking.

### DISCUSSION THREAD

**Discussion by bitbound** [microsoft/WindowsAppSDK#1723] [Q&A] [answered]
I'd like to use the `GraphicsCapturePicker` in a WinUI 3 Desktop app.   However, I get the following error when running this code:

```
    var isSupported = GraphicsCaptureSession.IsSupported(); // true
    var picker = new GraphicsCapturePicker();
    var item = await picker.PickSingleItemAsync();
```

```
System.Runtime.InteropServices.COMException: 'The application called an interface that was marshalled for a different thread. (0x8001010E (RPC_E_WRONG_THREAD))'
```

For WPF, I was able to get around this using the examples here: 
https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/tree/master/dotnet/WPF/ScreenCapture

But the same technique doesn't appear to be working with WinUI.  Specifically, marshalling the `DispatcherQueueController` here returns `null`: https://github.com/microsoft/Windows.UI.Composition-Win32-Samples/blob/master/dotnet/WPF/ScreenCapture/Composition.WindowsRuntimeHelpers/CoreMessagingHelper.cs

What do I need to do to get the `GraphicsCapturePicker` to work?

Thanks!  :)

**Comment by bitbound** [comment, marked as answer]
Nevermind.  I figured it out!  :)  Here's the interop calls that need to be made:

```
            var hwnd = WindowNative.GetWindowHandle(_mainWindow);
            var picker = new GraphicsCapturePicker();
            InitializeWithWindow.Initialize(picker, hwnd);
            var item = await picker.PickSingleItemAsync();
```

**Reply by OxOld** [reply]
nice

**Reply by codedocu** [reply]
great... thank you

**Reply by Fygadi** [reply]
Thx a lot u helped me too!

**Reply by kyleishie** [reply]
Thank you!
